### DISCUSSION TITLE
Impact of number of logging variables on Crazyradio packets

### DISCUSSION THREAD

**Discussion by NageshEranki** [bitcraze/discussions#1269] [Q&A]
Hi all,

I ran a small experiment to see the effects of increasing the number of logging variables on available radio channel capacities for mocap localization.  As of today, one logs through 'logging configurations' that listen for variable updates as and when the CF decides to send them. I wondered if logging too many variables can reduce the frequency of localization packets I can send to the CF.

As a baseline, I logged in one experiment, 3 variables: ```['stateEstimate.x','stateEstimate.y,'stateEstimate.z']```. Then, using Wireshark, I sniffed the Crazyradio packets and looked at the histogram of time-deltas between successive ```port:6, channel:1``` packets. 

I then repeated the experiment but this time logged 12 variables ```['stateEstimate.x','stateEstimate.y,'stateEstimate.z',,'stateEstimate.roll', ,'stateEstimate.pitch','stateEstimate.yaw','stateEstimate.vz','stateEstimate.vx','stateEstimate.vy','gyro.x','gyro.y',gyro.z']```

Interestingly, I saw no change in the time-deltas as shown in the histograms below:

![pcap_test1](https://github.com/bitcraze/discussions/assets/28665953/1efc9c79-bb11-4280-bb4d-eb15acaa7e4e)

![pcap_test2](https://github.com/bitcraze/discussions/assets/28665953/655c8b02-55fb-4a76-886f-0f337bb38c84)

Is this expected or do could I have a bug in my set-up?

Thanks

**Comment by ataffanel** [comment]
This sounds correct: the radio performance can be seen more in terms of packet per seconds than in net data bandwidth. This is due both to the current USB implementation that makes the transfer of each packet synchronous on a USB ping-pong and to the fact that you always have a fix cost of setting up the radio, sending preamble and receiving ack. This fix cost becomes big for small data packets.

So that you are seeing, where sending big packet is only marginally worst than sending large packet, in terms of packets per seconds, is exactly what I would expect. This is then important to send packets that are as big as possible if bandwidth is important.
